packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+
+  metrics?: { // Set visibility for telemetry metrics. Nested object future-proofing, but could flatten.
+      visibility: 'private' | 'org' | 'updateGroups' | 'public'  ; // TODO: should we share with specific groups?
+  };
+
+  contentDisplay?: 'thumbnail' | 'map'; // [Future] View configuration options such as cartography, charts, table, etc.
+  source: IHubSocial; // [Future] each of these has common metadata like `title`, `thumbnail` , and `link`
+  
+  // Content specific values. Combination of relevant item.data, layer info, enrichments, configuration settings
+  // could use values instead - which is common within item.data.values
+  attributes?: {
+      // Dataset, e.g.
+      fields?: [{name:string, type:string, statistics:Array<any>}];
+      recordNumber?: number;
+      // Map, e.g.
+      layers?: [];
+      basemap?: string;
+      // Document
+      format?: 'link' | 'PDF' | 'MS Word' | 'MS Excel' | 'Text';
+      // Initiative
+      stage?: string;
+      followersNumber?: number;
+      coreTeamId?: string; 
+      contentTeamId?: string;
+      supportingTeamsNumber?: number;
+      // Event
+      sponsors?: Array<any>;
+      attendeesNumber?: number;
+      limitNumber?: number;
+      startDate?: Date;
+      endDate?: Date;
+      videoUrl?: string;
+  };
+}
+
+// Common interface to people networks: Teams, Orgs, Members
+export interface IHubSocial {
+  id: string;
+  name: string;
+  thumbnailUrl?: string;
+  region: string; // user.region - US/Europe/etc.
+  culture: string; // e.g. user.culture
+  createdDate: Date;
+  updatedDate: Date;
+  url: string;
+}
+export interface IHubOrg extends IHubSocial { }
+
+export interface IHubTeam extends IHubSocial  { }
+
+// Simple user info - more could be added/cached
+export interface IHubMember extends IHubSocial {
+  username: string; // user.username
+  thumbnailUrl?: string; // user.thumbnail
+}
+
+// title, description, and optional link to license item with more info
+export interface IHubLicense {
+  id: string; // Licenses may be Items, so provide reference for links
+  title: string;
+  description: string;
+  link: string;
+  thumbnailUrl: string;
+  capabilities: {
+    reuse: boolean;
+    commercial: boolean;
+    derivatives: boolean;
+    attribution: boolean;
+    osm: boolean;
+  }
+}
+
+export interface IHubGeography {
+  title:string;
+  coverage?: 'global' | 'regional' | 'local'; // enrichment
+  geometry?: string; // serialized JSON, or should this refer to /resources/boundary.json
+  source: string; // item ID used for setting geometry
+}
+
+// Optional configured app links that replace "Create StoryMap" with links to specific apps/sites
+// per https://esriarlington.tpondemand.com/entity/96316-content-viewer-sees-associated-app-links
+// 'home' would be special as the landing page for the item instead of /content/:id/about or ../explore
```

Comment:
You mean someone might supply { title: 'home', url: 'https://some.example.org/path/to/page' }` and the UI would know that b/c the title is `'home'` that's what it would use for the result's link in search results and such?

Reply:
@tomwayson that's what I'm suggesting. It's perhaps too implied that `'home'` is a well-known value. Instead, we could instead later add a more explicit `content.home = ...`